pkg/apis/datavolumecontroller/v1alpha1/types.go

```diff
@@ -42,9 +42,10 @@ type DataVolumeSpec struct {
 
 // DataVolumeSource represents the source for our Data Volume, this can be HTTP, S3 or an existing PVC
 type DataVolumeSource struct {
-	HTTP *DataVolumeSourceHTTP `json:"http,omitempty"`
-	S3   *DataVolumeSourceS3   `json:"s3,omitempty"`
-	PVC  *DataVolumeSourcePVC  `json:"pvc,omitempty"`
+	HTTP   *DataVolumeSourceHTTP `json:"http,omitempty"`
+	S3     *DataVolumeSourceS3   `json:"s3,omitempty"`
+	PVC    *DataVolumeSourcePVC  `json:"pvc,omitempty"`
+	UPLOAD *bool                 `json:"upload,omitempty"`
```

Comment:
I think we should follow the pattern with other sources and introduce a DataVolumeSourceUpload type.  For now this type can be an empty struct.  This will allow us to extend the functionality more easily in the future.

Reply:
how the upload-datavolume.yaml should look like if we define it as an empty struct?
the code in datavolume-controller.go check if the struct is not null before setting the annotation.
How can you define it in the yaml as not null but not setting any field inside?